cpp/ql/lib/semmle/code/cpp/models/implementations/Strcpy.qll

```diff
@@ -110,8 +117,9 @@ class StrcpyFunction extends ArrayFunction, DataFlowFunction, TaintFunction, Sid
     exists(this.getParamSize()) and
     input.isParameterDeref(this.getParamSrc()) and
     (
-      output.isParameterDeref(this.getParamDest()) or
-      output.isReturnValueDeref()
+      output.isParameterDeref(this.getParamDest())
+      or
+      not this.returnsTotalLength() and output.isReturnValueDeref()
```

Comment:
Probably this should even be:
```suggestion
      or
      not this.returnsTotalLength() and output.isReturnValueDeref()
      or
      this.returnsTotalLength() and output.isReturnValue()
```

Reply:
I'm not even sure if we want to taint the return value when it's returning a length. In https://github.com/github/codeql/pull/3592 we removed taint flow from `strlen` because it apparently gave too many FPs, and it may be the case that it's the same thing here 🤔

Reply:
In that case is should probably be removed from the `strlcat` model

Reply:
Yeah, I think I agree

Reply:
Done in https://github.com/github/codeql/pull/14735/commits/1c878750494e87603a8519b5463297cbfb3b6712